src/gsi-ncdiag/gsi_ncdiag.py

```diff
@@ -720,24 +732,34 @@ def toIODAobs(self, OutDir, clobber=True, platforms=None):
                     continue
                 print("Platform:%s Var:%s #Obs:%d" % (p, v, np.sum(idx)))
 
-                writer = iconv.NcWriter(outname, LocKeyList, TestKeyList=TestKeyList)
-
                 outvars = conv_varnames[v]
                 for value in outvars:
-                    varDict[value]['valKey'] = value, writer.OvalName()
-                    varDict[value]['errKey'] = value, writer.OerrName()
-                    varDict[value]['qcKey'] = value, writer.OqcName()
+                    varDict[value]['valKey'] = value, iconv.OvalName()
+                    varDict[value]['errKey'] = value, iconv.OerrName()
+                    varDict[value]['qcKey'] = value, iconv.OqcName()
+                    VarDims[value] = ['nlocs']
+                    varAttrs[varDict[value]['valKey']]['units'] = units_values[value]
+                    varAttrs[varDict[value]['errKey']]['units'] = units_values[value]
+                    varAttrs[varDict[value]['qcKey']]['units'] = 'unitless'
+                    varAttrs[varDict[value]['valKey']]['coordinates'] = 'longitude latitude'
+                    varAttrs[varDict[value]['errKey']]['coordinates'] = 'longitude latitude'
+                    varAttrs[varDict[value]['qcKey']]['coordinates'] = 'longitude latitude'
+                    varAttrs[varDict[value]['valKey']]['_FillValue'] = self.FLOAT_FILL
+                    varAttrs[varDict[value]['errKey']]['_FillValue'] = self.FLOAT_FILL
+                    varAttrs[varDict[value]['qcKey']]['_FillValue'] = self.INT_FILL
 
                 for o in range(len(outvars)):
                     obsdata = self.var(conv_gsivarnames[v][o])[idx]
                     if outvars[o] == 'surface_pressure':
-                        if np.max(obsdata) < 1100.:
+                        if np.median(obsdata) < 1100.:
                             obsdata = obsdata * 100.  # convert to Pa from hPa
                     obserr = self.var('Errinv_Input')[idx]
                     mask = obserr < self.EPSILON
                     obserr[~mask] = 1.0 / obserr[~mask]
-                    obserr[mask] = self.FLOAT_FILL
-                    obserr[obserr > 4e8] = self.FLOAT_FILL
+                    # below is a temporary hack until missing ObsError support returns to IODA/UFO
```

Comment:
Is this still an issue? The ioda reader should be honoring the fill values in ObsError variables, but I'm not sure about UFO processing missing values in ObsError variables.

If it is an issue, the hack is just putting in a large value in ObsError so the corresponding ObsVaule has insignificant impact. Is this correct?

Reply:
Yes, perhaps this is fixed now but this is something @gthompsnJCSDA and I were testing a few months back. IODA/UFO complained about missing ObsError values. I think it is likely a UFO issue, not IODA, but perhaps Greg can comment more?

Reply:
I currently do not know if the incoming obsError is set to missing whether UFO ignores the ob.  Wojciech recently commented to me that it should be fine to have a missing incoming value and letting the ```filters``` apply the value, but I haven't tested it that way yet.  So for the purposes of our 1-month test dataset, I think we might want to keep it the way it was the last time we discussed this topic.